atlasdb-feign/build.gradle

```diff
@@ -0,0 +1,64 @@
+apply from: "../gradle/publish-jars.gradle"
+apply plugin: 'org.inferred.processors'
+apply plugin: 'com.github.johnrengelman.shadow'
+
+apply from: '../gradle/shared.gradle'
+
+configurations {
+    explicitShadow
+    compile.extendsFrom(explicitShadow)
+    shadow.extendsFrom(explicitShadow)
+
+    // This is used for cleaning up the POM.
+    toBeShaded
+    compile.extendsFrom(toBeShaded)
```

Comment:
Good question. It's a bit messy:

* `shadow` refers to dependencies that we expect to be available on the classpath at runtime, if one is running with the shadow JAR.
* `compile` refers to dependencies needed at compile time.
* `explicitShadow` is a thing we use for readability to mean "needed at both compile time and runtime-with-shadow-JAR" - the shadow configuration doesn't automatically inherit from `compile`. Usually this is for dependencies we are not shading.
* `toBeShaded` is purely for readability (I find 'compile' meaning 'not in the shadow jar at runtime' pretty unintuitive, especially since normal runtime inherits from compile). This would be functionally unchanged if we just used `compile`.